subsys/bluetooth/mesh/time_srv.c

```diff
@@ -504,6 +512,16 @@ int _bt_mesh_time_srv_update_handler(const struct bt_mesh_model *model)
 		return err;
 	}
 
+	/* If sent as an unsolicited message, the Time Status message shall be sent
+	 * with TTL=0 to avoid building up cumulative time errors resulting from delays
+	 * in processing the messages by relays.
+	 */
+	if (!bt_mesh_model_pub_is_retransmission(pub->mod)) {
+		cached_ttl = pub->ttl ? pub->ttl : cached_ttl;
+		pub->ttl = is_unsolicited ? 0 : cached_ttl;
```

Comment:
`bt_mesh_time_srv_time_status_send` calls `send_time_status` which calls `bt_mesh_msg_send`. `bt_mesh_msg_send` uses `bt_mesh_model_send` instead of `bt_mesh_model_publish` when message context is provided:
https://github.com/zephyrproject-rtos/zephyr/blob/080d51bb367836b73021fdd7ade3e2c4c54a0435/subsys/bluetooth/mesh/msg.c#L97-L99
It may happen that a user calls `bt_mesh_time_srv_time_status_send` with non-NULL `ctx`, which will set `is_unsolicited` to `true`. In this case the message will be passed to the transport layer directly bypassing the publication mechanism, and this callback won't be triggered. Then later if periodic publication is configured this callback will be called, but `is_unsolicited` will be set to `true` which in turns will set `ttl` to zero. I guess this is not what we want here.